naming/src/main/java/com/alibaba/nacos/naming/consistency/ephemeral/distro/v2/DistroClientDataProcessor.java

```diff
@@ -166,16 +168,33 @@ private void upgradeClient(Client client, ClientSyncData clientSyncData) {
         List<String> groupNames = clientSyncData.getGroupNames();
         List<String> serviceNames = clientSyncData.getServiceNames();
         List<InstancePublishInfo> instances = clientSyncData.getInstancePublishInfos();
+        List<BatchInstancePublishInfo> batchInstancePublishInfos = clientSyncData.getBatchInstancePublishInfos();
         Set<Service> syncedService = new HashSet<>();
         for (int i = 0; i < namespaces.size(); i++) {
             Service service = Service.newService(namespaces.get(i), groupNames.get(i), serviceNames.get(i));
             Service singleton = ServiceManager.getInstance().getSingleton(service);
             syncedService.add(singleton);
-            InstancePublishInfo instancePublishInfo = instances.get(i);
-            if (!instancePublishInfo.equals(client.getInstancePublishInfo(singleton))) {
-                client.addServiceInstance(singleton, instancePublishInfo);
-                NotifyCenter.publishEvent(
-                        new ClientOperationEvent.ClientRegisterServiceEvent(singleton, client.getClientId()));
+            InstancePublishInfo clientInstancePublishInfo = client.getInstancePublishInfo(singleton);
+            if (CollectionUtils.isNotEmpty(batchInstancePublishInfos)) {
```

Comment:
What's about when synced batchInstancePublishInfos last time but no batchInstancePublishInfos next time?

Reply:
> What's about when synced batchInstancePublishInfos last time but no batchInstancePublishInfos next time?

I fix it, the empty instance is judged in else, and the 400 status code is returned to the client in the handler

Reply:
I think about it again, we not support the deregister batch, if we not allow users batchRegister empty list, how to remove these instance?